### DISCUSSION TITLE
"Decouple" sessions from the identities?

### DISCUSSION THREAD

**Discussion by jonnyijapan** [ory/kratos#2546] [Q&A] [answered]
It seems that Kratos gives us the "identities" feature. However it seems it is coupled to sessions.
What I mean by that is it would be handy to only make use of the "identities" feature of Kratos, and let us use our own session feature.

What I am envisioning is:

## account creation
[web user creates new account, entering email/pw] > [our web app receives the email/pw] > [web app sends the email/pw to kratos service] > [kratos responds with account created] > [web app also sends a login action to kratos app] > [kratos responds with login accepted] > [web app (not kratos) creates the session] > [user is logged in]

Later, used tries to login on another device etc.

## account login
[web user logs in, entering email>pw] > [our web app receives email/pw] > [our web app receives the email/pw] > [web app sends the email/pw to kratos service] > [kratos responds with login accepted] > [web app (not kratos) creates the session] > [user is logged in]

Basically we'd like to put Kratos only "behind" our web backend layer, such that web user never interacts directly with Kratos service, no reverse proxy, etc. In our web app backend we'd only communicate with Kratos using Kratos APIs.

Maybe this is already possible with Kratos?

The reason is, I actually already have a working session model and don't need (cannot) reimplement it.

**Comment by jonnyijapan** [comment, marked as answer]
I see wording in the Kratos docs like
- Bring your own Identity Schema(s)
- Bring your own User Interface (Framework)

These are all very nice of course. Maybe what I would like is "Bring your own (web) sessions".

**Reply by jonnyijapan** [reply]
I've been playing with Kratos one or two days now, and have been able to create accounts (I should say "identity" I guess) using the "self-service" API. And I have also been able to log a user in using the two APIs "initialize login flow", then "submit a login flow". 

It seems like the way I want to use this should be pretty much possible. It seems like I don't have to surface Kratos' sessions onto the frontend.
Though Kratos sessions still of course seem to exist, and by default it seems like those run for like a month before they time out. Maybe I can extend those silently on the backend.

**Reply by aeneasr** [reply]
Awesome, great that you figured it out!

**Reply by vinckr** [reply]
Hello @jonnyijapan 
This discussion is quite old, but how has it been going since then? 
I will mark it as answered since your problem seems to be resolved, but feel free to come back to this discussion anytime, 
Thanks
